test/softmax.jl

```diff
@@ -1,12 +1,12 @@
 @testset "softmax" begin
-    for dims in [(5,5), (5,)]
-        x = randn(Float64, dims)
-        y = softmax(x)
-        dy = randn(Float64, dims)
-        gputest(softmax, x)
-        gputest(∇softmax, dy, x, y, checkgrad=false)
-        y = logsoftmax(x)
-        gputest(logsoftmax, x)
-        gputest(∇logsoftmax, dy, x, y, checkgrad=false) 
+    for (sz, dims) in [((5,), :), ((5,), 1), ((5,5), :), ((5,5), 1), ((5,5), 2)]
```

Comment:
Might be worth adding a few more:
```suggestion
    for (sz, dims) in [((5,), :), ((5,), 1), ((5,5), :), ((5,5), 1), ((5,5), 2), ((5,5,5,5), (2,3)), ((5,5,5,5), (2,4))]
```
because there's some clever reshaping, and a different path if `dims` have gaps in `lo:hi`.

https://github.com/FluxML/NNlibCUDA.jl/blob/master/src/cudnn/softmax.jl#L46